app/mailers/user_mailer.rb

```diff
@@ -48,15 +46,79 @@ def work_collaborator(user, work)
 
   def nightly_user_activity(user)
     @user = user
-    user_activity(@user)
-    if @active_user
+    @user_activity = Activity.build(user)
+
+    if @user_activity.has_activity
       mail to: @user.email, subject: "New FromThePage Activity"
     end
   end
 
   private
+
   def add_inline_attachments!
     attachments.inline["logo.png"] = File.read("#{Rails.root}/app/assets/images/logo.png")
   end
 
+
+  class Activity
+    attr_accessor :added_works, :active_pages, :active_translations, :active_note_pages, :has_activity
+
+    def initialize(added_works:, active_pages:, active_translations:, active_note_pages:, has_activity:)
+      @added_works = added_works
+      @active_pages = active_pages
+      @active_translations = active_translations
+      @active_note_pages = active_note_pages
+      @has_activity = has_activity
+    end
+
+    class << self
```

Comment:
@benwbrum I chose to go with a `build` method instead of doing the work in the `initialize` so that we can instantiate it independently of all of that `build` logic if we have to. I have a feeling this will come in handy for mock/stub testing purposes.